### PR TITLE
Makes it so you cant teleport into transition edges.

### LINKED ISSUE #6216: Teleportation can send to unreachable tiles
You can't reach the 2 edge tiles of a z-level with normal movement because you will be sent to another z-level, but you can reach them with teleportation. Items and mobs in those edges are not easily retrievable (you'll need either telescience or TK, and even worse for mobs who can send themselves deliberately to prevent themselves from being killed easily).

This can ruin nukeops and rev.

### PR DESCRIPTION
**Pull request by nullbear**
Also adds a helper proc for checking whether or not an atom is in the
transition edges of the map.
Fixes #6216

### CODE REVIEW

**Comment by Razharas** [comment]
Instead of checking the zlevel bounds just check if the turf have transition coordinates defined

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -323,3 +323,9 @@
 	SSair.remove_from_active(T0)
 	T0.CalculateAdjacentTurfs()
 	SSair.add_to_active(T0,1)
+
+/turf/proc/is_transition_turf()
+	if(istype(src, /turf/open/space))
```
Comment (MrStonedOne) [inline code/game/turfs/turf.dm]: Have this proc return FALSE then define a space version of the proc that does the check then returns TRUE.

Never istype() src.

**Comment by phil235** [comment]
Don't name your PR "Fixes #6216" , the name should always explain what the PR actually does.

**Comment by MrStonedOne** [comment]
>Don't name your PR "Fixes ####" , the name should always explain what the PR actually does.

**Comment by optimumtact** [comment]
Don't empty quote another persons comment just thumbs it up with the reaction emoji

**Comment by nullbear** [comment]
uhm... phil, i have no idea what to name it then. Cause that's what it does.

**Comment by nullbear** [comment]
oops

**Pull request merged**